### PR TITLE
#104 monitoring group mvvm

### PR DESCRIPTION
**Pull request by refatK**
**View Changes only from this branch:** https://github.com/Landry333/Big-Owl/pull/129/files/12f958e8baf70572c3bf6d53f9d3e8563d256629..3ab146ae4c4855204408d8a9859105ac40fc27b2

-----

**Related Issues**
#104 

**Description**
The MonitoringGroupPageActivity was refactored to use MVVM style. The logic is the same, but separated to repository for db calls and ViewModel for data handling. I only made a UI change where the groups supervisor name was shown. This was unnecessary as the user monitoring the group is the user of the app.

Also MonitoringGroupListActivity was untouched as it is not used in the application anyways and will be removed in a different branch,

**How to test**
Please run the app and check that MonitoringGroup works as expected. If the user hasno group or a empty group, there shold be a pop up. If the user does have a group of users, you should be able to see the group name, the list of users and search on the users, as before

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public MonitoringGroupPageViewModel() {
+        authRepository = new AuthRepository();
+        groupRepository = new GroupRepository();
+        userRepository = new UserRepository();
+    }
+
+    public LiveData<Group> getGroup() {
+        if (selectedGroup == null) {
+            selectedGroup = new MutableLiveData<>();
+            loadGroup();
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java:33]: Really smart of you to put loadGroup into the method. It's less code for the activity classes.

File: app/src/main/java/com/example/bigowlapp/repository/Repository.java [hunk 2]
```diff
@@ -131,6 +132,31 @@ public boolean removeDocument(String docUId) {
         return tData;
     }
 
+    public MutableLiveData<List<T>> getDocumentsByUId(List<String> uIdList, Class<? extends T> tClass) {
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/repository/Repository.java:135]: I would rename this method to getDocumentsByListOfUId for more clarity.

The current name implies that you could get multiple documents using a single UId. Just a suggestion if you think it's not necessary to change, let me know.

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]
Overall really good job.

I just requested change involving the name of a method.

File: app/src/main/java/com/example/bigowlapp/repository/Repository.java (hunk 2 above)
Comment (refatK) [reply app/src/main/java/com/example/bigowlapp/repository/Repository.java:135]: Makes sense, will do

File: app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java (hunk 1 above)
Comment (refatK) [reply app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java:33]: I just did it how it was done in Android documentation:
https://developer.android.com/topic/libraries/architecture/viewmodel#implement

**Comment by refatK** [comment]
@ShawnGuerra-Bautista did your requested changed

File: app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java [hunk 3]
```diff
@@ -1,147 +1,82 @@
...
-                                                })
-                                                .setCancelable(false)
-                                                .create()
-                                                .show();
-                                    }
-                                }
-                            }
-                        });
+
+        mGroupPageViewModel.getGroup().observe(this, group -> {
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java:49]: Very clean and good way to use ViewModel!

File: app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import com.example.bigowlapp.model.Group;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.GroupRepository;
+import com.example.bigowlapp.repository.UserRepository;
+
+import java.util.List;
+
+public class MonitoringGroupPageViewModel extends ViewModel {
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java:15]: A very useful view model which can be easily used by developers! nice!

**Review by KhangZhengNg — APPROVED** [review]

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Review by darkheros12 — APPROVED** [review]
The change are excellent. This is make our testing easier and the structure is more modularized.

**Comment by refatK** [comment]
Branch failed CI due to coverage, but that will be added in later PR.

**Pull request merged**